File: golang/escape_html.go
package nodx

import "strings"

// escapeHTML escapes the input string to prevent XSS attacks.
func escapeHTML(input string) string {
	replacements := map[string]string{
		"&":  "&#38;",
		"<":  "&#60;",
		">":  "&#62;",
		"\"": "&#34;",
		"'":  "&#39;",
	}

	var builder strings.Builder

	for _, char := range input {
		strChar := string(char)
		if replacement, exists := replacements[strChar]; exists {
			builder.WriteString(replacement)
		} else {
			builder.WriteString(strChar)
		}
	}

	return builder.String()
}

File: typescript/src/escape_html.ts
// escapeHTML escapes the input string to prevent XSS attacks.
export function escapeHtml (unsafe: string): string {
  return unsafe
    .replace(/&/g, '&#38;')
    .replace(/</g, '&#60;')
    .replace(/>/g, '&#62;')
    .replace(/"/g, '&#34;')
    .replace(/'/g, '&#39;')
}
